keel-core/src/main/kotlin/com/netflix/spinnaker/keel/api/ConstraintState.kt

```diff
@@ -0,0 +1,34 @@
+package com.netflix.spinnaker.keel.api
+
+import java.time.Duration
+import java.time.Instant
+
+data class ConstraintState(
+  val environmentUid: String,
+  val artifactVersion: String,
+  val constraintType: String,
+  val status: ConstraintStatus,
+  val createdAt: Instant = Instant.now(),
+  val judgedBy: String? = null,
+  val judgedAt: Instant? = null,
+  val comment: String? = null,
+  val json: String? = null
+) {
+  fun canPromote() = status.passes()
+
+  fun failed() = status.failed()
+
+  fun timedOut(timeout: Duration, now: Instant) =
+    createdAt.plus(timeout).isBefore(now)
```

Comment:
`isAfter(now)`?

Reply:
if `createdAt.plus(timeout)` is after `now`, the constraint hasn't timed out.